src/audio/Audio.js

```diff
@@ -58,23 +62,19 @@ THREE.Audio.prototype.setNodeSource = function ( audioNode ) {
 
 };
 
-THREE.Audio.prototype.setBuffer = function ( audioBuffer ) {
+THREE.Audio.prototype.setBuffer = function( audioBuffer ) {
 
 	var scope = this;
 
-	audioBuffer.onReady( function( buffer ) {
-
-		scope.source.buffer = buffer;
-		scope.sourceType = 'buffer';
-		if ( scope.autoplay ) scope.play();
-
-	} );
+	scope.source.buffer = audioBuffer;
+	scope.sourceType = 'buffer';
+	if ( scope.autoplay ) scope.play();
 
 	return this;
 
 };
 
-THREE.Audio.prototype.play = function () {
```

Comment:
I'll change that now. Sorry for these little mistakes. It is my first time contributing :).

I have tried to follow the contribution guidelines though and used http://zz85.github.io/mrdoobapproves/ which is why I removed the spaces, is it an issue with the site?

![image](https://cloud.githubusercontent.com/assets/820781/13860040/4bbb1b4a-ec7e-11e5-93d2-13ba91fca728.png)